nixos/lib/default.nix

```diff
@@ -0,0 +1,46 @@
+let
+  # The warning is in a top-level let binding so it is only printed once.
+  inherit (import ../../lib) warn;
+  experimentalWarning = warn "lib.nixos.evalModules is experimental and subject to change. See nixos/lib/default.nix" null;
+in
+{ lib ? import ../../lib, ... }:
+let
+
+  /*
+    Invoke NixOS. Unlike traditional NixOS, this does not include all modules.
+    Any such modules have to be explicitly added via the `modules` parameter,
+    or imported using `imports` in a module.
+
+    A minimal module list improves NixOS evaluation performance and allows
+    modules to be independently usable, supporting new use cases.
+
+    Parameters:
+
+      modules:        A list of modules that constitute the configuration.
+
+      specialArgs:    An attribute set of module arguments. Unlike
```

Comment:
Maybe also add to the description that these are not meant for general use and should only be used when _module.args can't be used.

Reply:
Now it's

```
      specialArgs:    An attribute set of module arguments. Unlike
                      `config._module.args`, these are available for use in
                      `imports`.
                      `config._module.args` should be preferred when possible.
```

Reply:
> `config._module.args` should be preferred when possible.

Doing more in from inside the modules is good I suppose and I think maybe it could be typed this way as well.

What would @Infinisil, who I believe came up with this recommendation, tell the reader?